### ISSUE TITLE
Tags with a space in the name = several modules say that there is a change even when there is no change

### ISSUE THREAD

**Issue by KortexSys** [ansible/ansible_collections_azure#409]
Hi,

I'm starting to use the Ansible modules provided by Microsoft to manage my Azure resources.

Some modules say there is a change even when there is no change.

For example :
  - I create Ansible code to create an Azure Traffic Manager Profile resource
  - I execute the code: my resource is created
  - I re-run the playbook (in check or normal) -> Ansible says there is a change.

I troubleshooted the problem to understand where is it.

In my case, the root cause was tags: I have tags with space in the name.

So if I do that it works:

```- name: "My resource"
  azure.azcollection.azure_rm_trafficmanager:
    api_profile: "latest"
    append_tags: "no"
    auth_source: "auto"
    cert_validation_mode: "validate"
    client_id: "my-client-id"
    cloud_environment: "AzureCloud"
    dns_config:
      relative_name: "my-name"
      ttl: "60"
    location: "global"
    monitor_config:
      path: "/"
      port: "443"
      protocol: "HTTPS"
    name: "my-name"
    profile_status: "Enabled"
    resource_group: "myResourceGroup"
    secret: "my-secret"
    state: "present"
    subscription_id: "my-subscription-id"
    tags: { "myExample1": "myValue1", "myExample2": "myValue2" }
    tenant: "my-tenant"
    traffic_routing_method: "Priority"
```

But if I do that it doesn't works and I have a "changed" each time I execute my playbook:
```- name: "My resource"
  azure.azcollection.azure_rm_trafficmanager:
    api_profile: "latest"
    append_tags: "no"
    auth_source: "auto"
    cert_validation_mode: "validate"
    client_id: "my-client-id"
    cloud_environment: "AzureCloud"
    dns_config:
      relative_name: "my-name"
      ttl: "60"
    location: "global"
    monitor_config:
      path: "/"
      port: "443"
      protocol: "HTTPS"
    name: "my-name"
    profile_status: "Enabled"
    resource_group: "myResourceGroup"
    secret: "my-secret"
    state: "present"
    subscription_id: "my-subscription-id"
    tags: { "my Example 1": "myValue1", "my Example 2": "myValue2" }
    tenant: "my-tenant"
    traffic_routing_method: "Priority"
```

--> because of the space in the tag name ("my Example 1" / "my Example 2").

I'm not sure but the problem is maybe in the update_tags method in the azure_rm_common.py file

Is it possible to develop a patch for this please :) ?
Because on Azure portal side it seems spaces are allowed in the tag name so I think it's a but :)

**Comment by KortexSys** [comment]
In fact I have a doubt about the update_tags method: it seems the problem is more in the "response".

When a module is executed, it retrieve informations about the remote resource to have all details about this resource.
In this response, it seems tags with a space in the name are missing.

**Comment by Fred-sun** [comment]
@brsysadmin Thanks for your contribute this issue! I checked it in the Azure Portal side. It's not support spaces in the tag name. Please recheck it! Thank you very mcuh!

**Comment by KortexSys** [comment]
> @brsysadmin Thanks for your contribute this issue! I checked it in the Azure Portal side. It's not support spaces in the tag name. Please recheck it! Thank you very mcuh!

How did you test please :) ?
Because I just realized a test on my side in the portal and I was able to insert a tag with a space in the name.
And so I don't know if we are talking about the same thing :)

<img width="810" alt="Azure_Tag_Space" src="https://user-images.githubusercontent.com/24649997/106752845-31333b00-662b-11eb-9242-3690309f35ab.png">

**Comment by Fred-sun** [comment]
@brsysadmin I added MyExample3 and My Example 4 to my portal at same time, but when saved, there is only MyExample3 in the current resource. Such as images. Thank you very much
![pic1](https://user-images.githubusercontent.com/37327967/106760009-5301e180-666e-11eb-92d4-7c74315ccea7.png)
![pic2](https://user-images.githubusercontent.com/37327967/106760015-54cba500-666e-11eb-9a2d-7bb0c2d8c314.png)

!

**Comment by KortexSys** [comment]
So I don't know why in your case you can't and why in my case I can but it seems space are allowed :)

For example for Azure Resource Manager : 
- source: https://docs.microsoft.com/en-gb/azure/azure-resource-manager/management/tag-resources
- quote:
```Handling spaces

If your tag names or values include spaces, enclose them in double quotes.
```

Besides, when I tried to create my Azure Traffic Manager, I was able to set tags with space in the name.

But if I reexecute the same Ansible code, I have a "changed" because when the Ansible module request Azure to obtain details about the created resource, the tags with space in the name was not retrieved.

**Comment by Fred-sun** [comment]
@brsysadmin Please check the Note Note for this connection ---https://docs.microsoft.com/en-gb/azure/azure-resource-manager/management/tag-resources, currently Traffic Manager Resource does not allow Spaces. Thank you very much! 
```e.g.
 Note

Currently, Azure DNS zones and Traffic Manager services also don't allow the use of spaces in the tag.

Azure Front Door doesn't support the use of # in the tag name.

Azure Automation and Azure CDN only support 15 tags on resources.

**Comment by KortexSys** [comment]
Hi @Fred-sun 

I agree that there is this:
```Note

Currently, Azure DNS zones and Traffic Manager services also don't allow the use of spaces in the tag.

Azure Front Door doesn't support the use of # in the tag name.

Azure Automation and Azure CDN only support 15 tags on resources.
`

I hadn't read it: sorry :(

I just tested through the portal in a Traffic Manager (my bad, yesterday I tested in a virtual machine !): I can't create a tag with a space in the name.

But I continue to say there is a bug somewhere and maybe we just have to reconsider at which level :)

If I resume at this moment:
- I can create a tag with a space in the name with the Ansible module azure.azcollection.azure_rm_trafficmanager
- the same Ansible module (azure.azcollection.azure_rm_trafficmanager) doesn't retrieve tags with a space in the name
- I can't create a tag with a space in the name with the Azure Portal on an Azure Traffic Manager

So just maybe to improve:
- the Ansible documentation for azure.azcollection.azure_rm_trafficmanager
- the Ansible module azure.azcollection.azure_rm_trafficmanager to block tags with a space in the name

What do you think about?

**Comment by Fred-sun** [comment]
@brsysadmin Yes, I will bring up PR to add description to the documentation. Thank you very much!

**Comment by Fred-sun** [comment]
fixes by #415
